alembic/versions/20230706_c471f553249b_migrate_library_logo.py

```diff
@@ -0,0 +1,66 @@
+"""Migrate library logo
+
+Revision ID: c471f553249b
+Revises: 04bbd03bf9f1
+Create Date: 2023-07-06 19:37:59.269231+00:00
+
+"""
+import logging
+
+import sqlalchemy as sa
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "c471f553249b"
+down_revision = "04bbd03bf9f1"
+branch_labels = None
+depends_on = None
+
+log = logging.getLogger(f"palace.migration.{revision}")
+log.setLevel(logging.INFO)
+log.disabled = False
+
+
+def upgrade() -> None:
+    op.create_table(
+        "libraries_logos",
+        sa.Column("library_id", sa.Integer(), nullable=False),
+        sa.Column("content", sa.LargeBinary(), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["library_id"],
+            ["libraries.id"],
+        ),
+        sa.PrimaryKeyConstraint("library_id"),
+    )
+
+    prefix = "data:image/png;base64,"
+    connection = op.get_bind()
+    libraries = connection.execute("select id, short_name from libraries")
+
+    for library in libraries:
+        setting = connection.execute(
+            "select cs.value from configurationsettings cs "
+            "where cs.library_id = (%s) and cs.key = 'logo'",
+            (library.id,),
+        ).first()
+        if setting and setting.value:
+            log.info("Library {library.short_name} has a logo, migrating it.")
+            logo_str = setting.value
+
+            # We stored the logo with a data:image prefix before, but we
```

Comment:
Are we always just assuming it's png forevermore?

Reply:
Yes, I suppose we are. Since we convert everything to PNG, I figured this was okay. I could add a content_type column to the LibraryImages table if you think that would be a good change.